hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java

```diff
@@ -161,7 +161,10 @@ public TransactionContext startTransaction(
   @Override
   public long takeSnapshot() throws IOException {
     LOG.info("Saving Ratis snapshot on the OM.");
-    return ozoneManager.saveRatisSnapshot();
+    if (ozoneManager != null) {
+      return ozoneManager.saveRatisSnapshot();
+    }
+    return 0;
```

Comment:
Question: Here we are returning 0,(Should we return the lastAppliedIndex we are writing) How this will be used by Ratis?

Reply:
We do return the last applied index which is being stored on disk. The null check is for tests only (TestOzoneManagerRatisServer).
Ratis currently does not do anything with the returned value.